### PR TITLE
Integration test for Java plugins

### PR DESCRIPTION
**Pull request by danhermann**
Adds an integration test that exercises Java plugins with an input, filter, and output.

Relates to #8429.

### CODE REVIEW

**Comment by danhermann** [comment]
Jenkins, test this please

File: qa/integration/specs/java_api_spec.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    @logstash.start_background_with_config_settings(config_to_temp_file(@fixture.config), settings_dir)
+
+    # wait for Logstash to start
+    started = false
+    while !started
+    begin
+      sleep(1)
+      result = @logstash.monitoring_api.event_stats
+      started = !result.nil?
+      rescue
```
Comment (colinsurprenant) [inline qa/integration/specs/java_api_spec.rb]: cosmetic: indentation

**Comment by colinsurprenant** [comment]
LGTM - minor comment.

**Comment by danhermann** [comment]
Jenkins, test this please

**Comment by danhermann** [comment]
Fixed indentation. Thanks, @colinsurprenant!

**Pull request closed without merging**